.circleci/ecr_gc_docker/gc.py

```diff
@@ -164,51 +167,48 @@ def looks_like_git_sha(tag):
 
     # Keep list of image digests to delete for this repository
     digest_to_delete = []
-    print(repositoryName)
 
     for image in images(client, repo):
         tags = image.get("imageTags")
         if not isinstance(tags, (list,)) or len(tags) == 0:
             continue
-
-        tag = tags[0]
         created = image["imagePushedAt"]
         age = now - created
-        if any([
-                looks_like_git_sha(tag),
-                tag.isdigit(),
-                tag.count("-") == 4,  # TODO: Remove, this no longer applies as tags are now built using a SHA1
-                tag in ignore_tags]):
-            window = stable_window
-            if tag in ignore_tags:
-                stable_window_tags.append((repositoryName, tag, "", age, created))
-            elif age < window:
-                stable_window_tags.append((repositoryName, tag, window, age, created))
+        for tag in tags:
+            if any([
+                    looks_like_git_sha(tag),
+                    tag.isdigit(),
+                    tag.count("-") == 4,  # TODO: Remove, this no longer applies as tags are now built using a SHA1
+                    tag in ignore_tags]):
+                window = stable_window
+                if tag in ignore_tags:
+                    stable_window_tags.append((repositoryName, tag, "", age, created))
+                elif age < window:
+                    stable_window_tags.append((repositoryName, tag, window, age, created))
+            else:
+                window = unstable_window
+
+            if tag in ignore_tags or age < window:
+                if args.debug:
+                    print("Ignoring {}:{} (age: {})".format(repositoryName, tag, age))
+                break
         else:
-            window = unstable_window
-
-        if tag in ignore_tags:
-            print("Ignoring tag {}:{} (age: {})".format(repositoryName, tag, age))
-            continue
-        if age < window:
-            print("Not deleting manifest for tag {}:{} (age: {})".format(repositoryName, tag, age))
-            continue
-
-        if args.dry_run:
-            print("(dry run) Deleting manifest for tag {}:{} (age: {})".format(repositoryName, tag, age))
-        else:
-            print("Deleting manifest for tag{}:{} (age: {})".format(repositoryName, tag, age))
+            for tag in tags:
+                print("{}Deleting {}:{} (age: {})".format("(dry run) " if args.dry_run else "", repositoryName, tag, age))
```

Comment:
This error message seems a bit misleading, since you'll print it N times for how many tags there are, but actually only one "delete" operation is actually going to happen in the end

Reply:
But it's good in a sense since it'll give us all the tags that will be deleted through this process.

In the previous version a single digest getting deleted would delete multiple tags which was hidden by the fact that it only printed one tag.

Reply:
I agree that seeing all the tags that actually get deleted is good. So I would like the output to look like that. But I also think we should accurately report what is *actually* happening when the script gets run.